whirlpool/eventsocket.py

```diff
@@ -14,20 +14,42 @@
 
 RECV_MSG_MATCHER = re.compile("{(.*)}\\x00")
 
+# closing frame status codes for websocket.
+STATUS_NORMAL = 1000
+STATUS_GOING_AWAY = 1001
+STATUS_PROTOCOL_ERROR = 1002
+STATUS_UNSUPPORTED_DATA_TYPE = 1003
+STATUS_STATUS_NOT_AVAILABLE = 1005
+STATUS_ABNORMAL_CLOSED = 1006
+STATUS_INVALID_PAYLOAD = 1007
+STATUS_POLICY_VIOLATION = 1008
+STATUS_MESSAGE_TOO_BIG = 1009
+STATUS_INVALID_EXTENSION = 1010
+STATUS_UNEXPECTED_CONDITION = 1011
+STATUS_SERVICE_RESTART = 1012
+STATUS_TRY_AGAIN_LATER = 1013
+STATUS_BAD_GATEWAY = 1014
+STATUS_TLS_HANDSHAKE_ERROR = 1015
+STATUS_UNAUTHORIZED = 3000
+
+RECONNECT_TIMEOUT = 30
+GOING_AWAY_TIMEOUT = 60*5 - RECONNECT_TIMEOUT
+
 
 class EventSocket:
-    def __init__(self, url, access_token, said, msg_listener: Callable[[str], None]):
+    def __init__(self, url, auth:Auth, said, msg_listener: Callable[[str], None],fetch_data:Callable):
```

Comment:
```suggestion
    def __init__(self, url, auth: Auth, said, msg_listener: Callable[[str], None], conn_up_listener: Callable):
```
Lets call it conn_up_listener, to make it generic: from the Websocket PoV, it does not know about what the Appliance does, it simply exposes callbacks that the Appliance can act on.

conn_up_listener = connection up